### PR TITLE
Class based user api

### PR DESCRIPTION
**Pull request by westb2**
Some notes:
-I plan on implementing a general put method in the abstract classes, but it's going to be a bit more complicated
-All the table and field names are still bad. This will be updated when I am done refactoring the database, which is on a different branch.
-If this looks good I plan on implementing the entire API this way

### CODE REVIEW

File: src/users/ApiService.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import {ApiModel} from './ApiModel';
+var db:any     = require('../server/sweet-skills-database');
+var person:any = require('./../models/person');
```
Comment (paulSambolin) [inline src/users/ApiService.ts]: remove specifics from an abstract class (unused here anyway)

File: src/users/ApiModel.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  * getAll():any{
+      var status = 409;
+      var body = '';
+      var model = this;
+      try{
+          yield model.databaseTable.findAll().then(function(results:any){
+              status = 200;
+              body   = results;
+              model.logger.info("Users retrieved");
```
Comment (paulSambolin) [inline src/users/ApiModel.ts]: remove 3 user specific logging statements from this abstract class

Comment (paulSambolin) [inline src/users/ApiModel.ts]: or handle it more generically

File: package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "main": "app.js",
   "scripts": {
     "test": "./node_modules/.bin/jasmine-node spec",
-    "start": "sudo node dist/app.js",
+    "start": "sudo node --debug dist/app.js",
```
Comment (westb2) [inline package.json]: Removed this, uneccessary now anyway

File: src/users/ApiModel.ts (hunk 2 above)
Comment (westb2) [inline src/users/ApiModel.ts]: Now handled more generically

**Comment by paulSambolin** [comment]
If it's all tested it all looks good to me!  Nice work on the TS class

**Pull request merged**